source/validate_id.cpp

```diff
@@ -2678,9 +2677,11 @@ spv_result_t IdPass(ValidationState_t& _,
     }
   }
   if (result_id) {
```

Comment:
I don't think there is a need for this block to be here. Consider moving it inside RegisterInstruction(). Then there is also no need to pipeline internal_instruction.